File: Lectures/CSC207/207_L0101/w01/Balloon.py
class Balloon:

    def __init__(self, color):
        self.amount = 0
        self.capacity = 100
        self.popped = False
        self.color = color

    def getAmount(self):
        return self.amount

    def getCapacity(self):
        return self.capacity

    def getIsPopped(self):
        return self.popped

    def pop(self):
        '''Set amount and capacity to 0, and popped to True.'''
        self.amount = 0
        self.capacity = 0
        self.popped = True



        

    def inflate(self, a):
        '''If balloon is not popped, and a is a positive int, add
 given int a to amount, as long as total does not exceed capacity. If total
 is larger than capacity, pop balloon.'''
        
        if not self.popped and a > 0:
            self.amount = self.amount + a;
            if self.amount > self.capacity:
                self.pop()



    def __str__(self):
        s = "Amount: " + str(self.amount) + ", Capacity: " + str(self.capacity) + ", Popped: " + str(self.popped)
        return s

File: Lectures/CSC207/207_L0101/w01/Balloon.java
public class Balloon {

	// the instance variables get declared
	// outside of any method
	private int amount;
	private int capacity;
	private boolean popped;
	private String color;
	
	// the constructor has the same name as the class
	// no return type; not even void
	public Balloon(String color) {
		this.amount = 0;
		this.capacity = 100;
		this.popped = false;
		// this is like Python's "self"
		this.color = color;
	}
	
	public void setAmount(int amnt) {
		this.amount = amnt;
	}
	
	public int getAmount() {
		return this.amount;
	}
	
	public int getCapacity() {
		return this.capacity;
	}
	
	public boolean getIsPopped() {
		return this.popped;
	}
	
	public void pop() {
		this.amount = 0;
		this.capacity = 0;
		this.popped = true;
	}

	public void inflate(int a) {
		
		if ((this.popped == false) && (a > 0)) {
			this.amount += a;
			if (this.amount > this.capacity) {
				this.pop();
			}
 		}
	}
	
	@Override
	public String toString() {
		// complete this
		return "";
	}
}
